### PR TITLE
refactor(react-intl): update to use hook instead of injectintl

### PR DESCRIPTION
**Pull request by fcoronelmakingsense**
- Change injectintl for useIntl() hook the other is obsolete and don't need anymore

### CODE REVIEW

File: src/components/Login/LoginErrorAccountNotValidated.js [hunk 1]
```diff
@@ -1,46 +1,45 @@
...
-     * @param { import('../../services/pure-di').AppServices } props.dependencies
-     */
-    ({ intl, email, dependencies: { dopplerLegacyClient } }) => {
-      const [resentTimes, setResentTimes] = useState(0);
-      const [Captcha, verifyCaptcha] = useCaptcha();
+export const LoginErrorAccountNotValidated = InjectAppServices(
+  /**
+   * @param { Object } props
+   * @param { string } props.email
+   * @param { import('react-intl').InjectedIntl } props.intl
```
Comment (andresmoschini) [inline src/components/Login/LoginErrorAccountNotValidated.js]: This parameter does not exist anymore, right?

**Comment by cbernat** [comment]
Nice work @fcoronelmakingsense

**Comment by andresmoschini** [comment]
Great work @fcoronelmakingsense !

Just to learn a little more, why did they prefer a _hook_ (`useIntl`) over a _hoc_ (a high order component `injectIntl`)? Maybe you have a link where they talk about the advantages.

**Comment by andresmoschini** [comment]
There is a commentary here, but it does not talk about advantages or disadvantages, it seems to be almost the same.

`injectIntl` is not deprecated, right? I mean for our code-base style `useIntl` is nicer, but `injectIntl` is totally supported. Am I missing something?

**Comment by fcoronelmakingsense** [comment]
> There is a commentary here, but it does not talk about advantages or disadvantages, it seems to be almost the same.
> 
> `injectIntl` is not deprecated, right? I mean for our code-base style `useIntl` is nicer, but `injectIntl` is totally supported. Am I missing something?

Sorry i forgot to put the link. It's not deprecated you right. https://github.com/formatjs/react-intl/blob/master/docs/Upgrade-Guide.md#new-useintl-hook-as-an-alternative-of-injectintl-hoc

File: src/components/Login/LoginErrorAccountNotValidated.js (hunk 1 above)
Comment (fcoronelmakingsense) [reply src/components/Login/LoginErrorAccountNotValidated.js]: yes! i removed comment now

**Pull request merged**